backend/src/db.rs

```diff
@@ -173,6 +173,24 @@ impl DB {
         if redeemables.is_empty() {
             return Ok(RedeemStatus::MarketNotRedeemable);
         }
+
+        let settled = sqlx::query!(
+            r#"WITH fund_and_constituents AS (
+                SELECT ? AS fund_id
+                UNION ALL
+                SELECT constituent_id
+                FROM redeemable
+                WHERE fund_id = ?
+            ) SELECT id FROM market WHERE id IN (SELECT id FROM fund_and_constituents) AND settled_price IS NOT NULL"#,
+            fund_id,
+            fund_id
+        ).fetch_all(transaction.as_mut())
+        .await?;
```

Comment:
This returns all settled markets, here's why:

- `fund_and_constituents` has the column `fund_id`, not `id`
- in `(SELECT id FROM fund_and_constituents)`, id refers to `market.id` from the outside scope
- thus the id is always in the subquery

Reply:
Try something like:

```
SELECT EXISTS (
  SELECT 1 FROM market JOIN redeemable ON (market.id = fund_id or market.id = constituent_id) WHERE fund_id = ? AND settled_price IS NOT NULL
)
```

Reply:
(you would use this with `query_scalar` and get a boolean representing whether any of the markets is settled)